datadog_checks_base/tests/base/utils/discovery/test_discovery.py

```diff
@@ -21,6 +22,21 @@ def test_include_not_empty():
     assert mock_get_items.call_count == 1
 
 
+def test_include_patterns_are_compiled_internally():
+    mock_get_items = mock.Mock(return_value=['a', 'b', 'c', 'd', 'e', 'f', 'g'])
+    d = Discovery(mock_get_items, include={'a.*': None, 'b.*': None})
+    assert list(d.get_items()) == [('a.*', 'a', 'a', None), ('b.*', 'b', 'b', None)]
+    assert mock_get_items.call_count == 1
+    assert d._filter._compiled_include_patterns == {p: re.compile(p) for p in ['a.*', 'b.*']}
```

Comment:
You are refactoring and not adding new functionality. In that case no new unit tests should be needed, and the proof of that is that you are checking private properties of the class that might be refactored in the future and force you to change your unit tests then, when that shouldn't be necessary.

Reply:
It is very siilar to `test_include_not_empty`, but yes... I think we can keep it

Reply:
so similar that we can merge them together! Done. Thanks